src/runmode-dpdk.c

```diff
@@ -0,0 +1,1295 @@
+        SCReturnPtr(NULL, "void *");
+    }
+    retval = ConfigLoad(iconf, iface);
+    if (retval < 0) {
+        iconf->DerefFunc(iconf);
+        SCReturnPtr(NULL, "void *");
+    }
+
+    return iconf;
+}
+
+void DeviceInitPortConf(const DPDKIfaceConfig *iconf, const struct rte_eth_dev_info *dev_info,
+        struct rte_eth_conf *port_conf)
+{
+    *port_conf = (struct rte_eth_conf){
+            .rxmode = {
+                    .mq_mode = ETH_MQ_RX_NONE,
+                    .max_rx_pkt_len = RTE_ETHER_MAX_LEN,
+                    .offloads = 0, // turn every offload off to prevent any packet modification
+            },
+            .txmode = {
+                    .mq_mode = ETH_MQ_TX_NONE,
+                    .offloads = 0,
+            },
+    };
+
+    // configure RX offloads
+    if (dev_info->rx_offload_capa & DEV_RX_OFFLOAD_RSS_HASH) {
+        if (iconf->nb_rx_queues > 1) {
+            SCLogConfig("RSS enabled on %s for %d queues", iconf->iface, iconf->nb_rx_queues);
+            port_conf->rx_adv_conf.rss_conf.rss_key = rss_hkey;
+            port_conf->rx_adv_conf.rss_conf.rss_key_len = RSS_HKEY_LEN;
+            port_conf->rx_adv_conf.rss_conf.rss_hf = ETH_RSS_TCP | ETH_RSS_UDP;
+            port_conf->rxmode.mq_mode = ETH_MQ_RX_RSS;
+        } else {
+            SCLogConfig("RSS not enabled on %s", iconf->iface);
+            port_conf->rx_adv_conf.rss_conf.rss_key = NULL;
+            port_conf->rx_adv_conf.rss_conf.rss_hf = 0;
+        }
+    } else {
+        SCLogConfig("RSS not supported on %s", iconf->iface);
+    }
+
+    if (iconf->checksum_mode == CHECKSUM_VALIDATION_DISABLE) {
+        SCLogConfig("Checksum validation disabled on %s", iconf->iface);
+    } else if (dev_info->rx_offload_capa & DEV_RX_OFFLOAD_CHECKSUM) {
+        if (iconf->checksum_mode == CHECKSUM_VALIDATION_ENABLE &&
+                iconf->flags & DPDK_RX_CHECKSUM_OFFLOAD) {
+            SCLogConfig("IP, TCP and UDP checksum validation enabled and offloaded "
+                        "on %s",
+                    iconf->iface);
+            port_conf->rxmode.offloads |= DEV_RX_OFFLOAD_CHECKSUM;
+        } else if (iconf->checksum_mode == CHECKSUM_VALIDATION_ENABLE &&
+                   !(iconf->flags & DPDK_RX_CHECKSUM_OFFLOAD)) {
+            SCLogConfig("Suricata checksum validation enabled "
+                        "but not offloaded on %s",
+                    iconf->iface);
+        }
+    }
+
+    if (dev_info->tx_offload_capa & DEV_TX_OFFLOAD_MBUF_FAST_FREE) {
+        port_conf->txmode.offloads |= DEV_TX_OFFLOAD_MBUF_FAST_FREE;
+    }
+}
+
+int DeviceConfigureQueues(DPDKIfaceConfig *iconf, const struct rte_eth_dev_info *dev_info,
+        const struct rte_eth_conf *port_conf)
+{
+    int retval;
+    uint16_t mtu_size;
+    uint16_t mbuf_size;
+    struct rte_eth_rxconf rxq_conf;
+    struct rte_eth_txconf txq_conf;
+
+    iconf->pkt_mempool_array = SCCalloc(iconf->nb_rx_queues, sizeof(struct rte_mempool *));
```

Comment:
in v3 and v4 I have explained it is wise to use single or NUMA specific mempool. It is not correct to create mempool per port-queue pair. Please rework this as it is not correct.